### PR TITLE
Implement register_template_directory to Registry 

### LINKED ISSUE #11: ^ as else
Support 

```handlebars
{{#if ...}}
...
{{^}}
...
{{/if}}
```

### PR DESCRIPTION
**Pull request by gamebox**
Allow a caller to specify a directory path and an extension, then
register all files in that directory.

Closes #11

### CODE REVIEW

File: examples/render-cli.rs [hunk 1]
```diff
@@ -44,6 +44,7 @@ fn main() {
         .register_template_file(&filename, &filename)
         .ok()
         .unwrap();
+
```
Comment (sunng87) [inline examples/render-cli.rs]: It's better to remove these newline to keep the changeset clean.

File: src/error.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+use std::convert::From;
```
Comment (sunng87) [inline src/error.rs]: I guess this is an unused import?

File: src/registry.rs [hunk 3]
```diff
@@ -174,6 +177,43 @@ impl Registry {
...
+            _ => {
+                for p in dir_iter.filter_map(|e| e.ok()) {
+                    let tpl_path = p.path();
+                    let tpl_file_path = p.path().to_string_lossy();
+                    let tpl_name = &tpl_file_path[prefix_len..tpl_file_path.len() - tpl_extension.len()];
+                    let tpl_canonical_name = tpl_name.replace("\\", "/");
+                    self.register_template_file(&tpl_canonical_name, &tpl_path)?;
+                }
+
+                 Ok(())
```
Comment (sunng87) [inline src/registry.rs]: indent

**Comment by sunng87** [comment]
Thanks @gamebox. I just added some comments. And it would be great if you add test case and doc for this function.

**Comment by gamebox** [comment]
Working on tests, having some issues with tempdir/tempfile(no files are showing up).  I'll try to update this PR as soon as I can.

**Comment by sunng87** [comment]
Thank you @gamebox! It's in no hurry, take your time.

**Comment by gamebox** [comment]
Thanks, I'm having a blast here, as I think I've taught myself about 12 rust things in the span of working on this feature and writing the tests.  Almost done :-)

**Comment by sunng87** [comment]
Hope you enjoy the process :smile:

**Comment by gamebox** [comment]
New commits are up, test coverage should be pretty good(mostly happy path though).  Hope this looks good, and if not, happy to get some feedback.

**Comment by sunng87** [comment]
Thank you! I will look into this later as I'm working on a massive refactoring in #221

**Comment by gamebox** [comment]
In no hurry!  That looks like some serious work.  I'd offer help, but I'm still barely crawling along with understanding lifetimes and moves.

**Comment by sunng87** [comment]
Looks good for me! Thank you for this feature! Hope you enjoy Rust and this library.

**Pull request merged**